packages/button/Button.tsx

```diff
@@ -1,33 +1,44 @@
-import { h, Component, prop } from 'skatejs';
+import { h, Component, prop, ComponentProps } from 'skatejs';
 
-import { ColorType, cssClassForColorType, css } from '../_helpers';
+import {
+  ColorType,
+  cssClassForColorType,
+  css,
+  IntrinsicCustomElement,
+  IntrinsicBoreElement,
+  ClickEvent
+} from '../_helpers';
 
 import styles from './Button.scss';
 
-// public
-type ButtonProps = Props & EventProps;
-type Props = {
+export type ButtonProps = Props & Attrs & EventHandlers;
+export type Attrs = {
```

Comment:
it depends on Component ofc.
 Button "primary" api is via attributes, that is also what is tested in test. Also this "clearly" separates type definitions to common pattern.
`Props` - primary API, should be preferred over attributes whenever possible
`Attrs` - only if primary API is via attributes,
`Events` - event aliases which we emit, specifically for bore `h`
`EventHandlers` - event aliases which we emit, specifically for IntrinsicElement consumption within other Virtual Dom view renderers

Attrs and Events are specifically for bores `h` while intersection of all 3 is our intrinsic element API.

We can discuss it further ofc, this is just what I ended up with, after evaluation of all possible scenarios